.gitlab/ci/bucket-upload.yml

```diff
@@ -0,0 +1,165 @@
+.bucket-upload-rule:
+  rules:
+    - if: '$GCS_MARKET_BUCKET == "marketplace-dist"' # remove this after testing period
+      when: never # remove this after testing period
+    - if: '$BUCKET_UPLOAD == "true"'
+
+.start_tunnel: &start_tunnel
+  - section_start "Start Tunnel"
+  - echo "Open an ssh tunnel to the demisto servers and wait until the tunnels are established"
+  - echo "Add ssh configurations"
+  - |
+    if [ -z $INSTANCE_CREATED ]; then
+      echo "Skipping - instance was not created"
+    else
+      cat $SSH_CONFIGURATION >> ~/.ssh/config
+    fi
+  - echo "Open SSH Tunnel"
+  - Tests/scripts/open_ssh_tunnel.sh
+  - section_end "Start Tunnel"
+
+.check_user_permissions_to_upload_packs: &check_user_permissions_to_upload_packs
+  - section_start "Check User Permissions to Upload Packs" # if bucket upload and uploading to marketplace-dist
+  - |
+    if [[ -n "${BUCKET_UPLOAD}" || -n "${FORCE_BUCKET_UPLOAD}" ]] && [[ "$GCS_MARKET_BUCKET" == "$GCS_PRODUCTION_BUCKET" ]]; then
+      CONTENT_LEADERS=$(curl -sS "https://api.github.com/orgs/demisto/teams/content-leaders/members" -H "Authorization: token ${GITHUB_TOKEN}")
+      echo "recieved content leaders"
+      LEADER_NAMES=$(echo $CONTENT_LEADERS | jq -r ".[].login")
+      LEADER_NAMES=$(echo "${LEADER_NAMES}" "content-bot" )
+      if [[ -z "$GITLAB_USER_NAME" ]] || [[ -z "`echo $LEADER_NAMES | grep -w "$GITLAB_USER_NAME"`" ]]; then
+        echo -e "User '$GITLAB_USER_NAME' is not allowed to trigger this build, only one of:\n${LEADER_NAMES}"
+        exit 1
+      else
+        echo "User '${GITLAB_USER_NAME}' is allowed to upload packs / force upload packs."
+      fi
+    fi
+  - section_end "Check User Permissions to Upload Packs"
+
+.install_packs_in_server:
+  needs: ["create-instances"]
```

Comment:
We only want to create the `6.0` and `master` instances when the `create-instances` job runs in this pipeline, right? How are you doing that currently, setting the `IFRA_ENV_TYPE` appropriately in the schedule trigger env vars?

Reply:
created a job that extends it